### DISCUSSION TITLE
How to create a mutation for many to many self relationship ?

### DISCUSSION THREAD

**Discussion by krunaldodiya** [prisma/prisma#5441] [Q&A] [answered]
I am trying to make follow system but not sure how to create followUser(following_id: "some_id") ???

```prisma
model User {
  id                 String  @id @default(uuid())
  name               String?

  following User[] @relation(name: "UserFollows", references: [id])
  followers User[] @relation(name: "UserFollows", references: [id])

  @@map("users")
}
```

```ts
import { objectType } from 'nexus'

export const User = objectType({
  name: 'User',
  definition(t) {
    t.model.id()
    t.list.field('followers', {
      type: 'User',
      resolve: async (root, args, { prisma, user }) => {
       // how can i specify relationship for followers ???
      },
    })
    t.list.field('following', {
      type: 'User',
      resolve: async (root, args, { prisma, user }) => {
       // how can i specify relationship for following ???
      },
    })
    t.model.name()
  },
})
```

```ts
import { mutationField, nonNull, stringArg } from 'nexus'

export const followUser = mutationField('followUser', {
  type: 'String',
  args: { following_id: nonNull(stringArg()) },
  resolve: async (parent, { following_id }, { prisma, user }) => {
    // how can i create mutation for following a user ???
  },
})

```

**Comment by ryands17** [comment, marked as answer]
For the `User` object, you can directly use `t.model`:

```ts
const User = objectType({
  name: 'User',
  definition(t) {
    t.model.id()
    t.model.name()
    t.model.following()
    t.model.followers()
  },
})
```

As for the Mutation, you can add a follow to a given user as follows:

```ts
const followeUser = ns.mutationField((t) => {
  t.field('followUser', {
    type: 'User',
    args: {
      following_id: nonNull(idArg())
    },
    resolve(_parent, { following_id }, { prisma, user }) {
      return prisma.user.update({
        where: { id: user.id },
        data: { following: { connect: { id: following_id } } },
      })
    },
  })
})
```

**Reply by krunaldodiya** [reply]
That was a pretty simple, i almost google for few hours , Thank for saving me :)

**Reply by ekafyi** [reply]
Sorry for thread-jacking but my question has the exact same use case as this one. In the Mutation example above, we update a user's `following`. Do we need to manually update the followed user's `followers` field as well?

**Reply by ryands17** [reply]
Yes you have to update both sides.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
